### PR TITLE
feat(config): added support for sunricher ZV2835RAC-NF

### LINKED ISSUE #2570: Add Sunricher Knob Dimmer device
**Which device is missing?**

Sunricher Dimmer Knob. It appears to be the same to this one: https://devices.zwave-js.io/?jumpTo=0x0330:0x0200:0xd017:1.14 but just reports different product id hence isn't recognized.
I purchased them over here (rebranded): https://www.robbshop.nl/z-wave-draaidimmer-robb-smarrt 
Manual is at: https://www.robbshop.nl/media/73/6e/6e/1607613471/robbsmarrtz-wavedraaidimmermanual_561490868.pdf

**What are the IDs?**

Manufacturer ID: 0x0330  
Product type: 512 (0x0200)  
Product ID: 53271 (0xd017)

**Did you check if we really don't have it?**

- [X] It is not in the [configuration DB](https://devices.zwave-js.io/)
- [X] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)
- [X] It is not in the [TODO list](https://github.com/zwave-js/node-zwave-js/issues/1600)

**Do you have a link to the PDF manual?**

Yes: https://www.robbshop.nl/media/73/6e/6e/1607613471/robbsmarrtz-wavedraaidimmermanual_561490868.pdf
If so, it helps in creating the config file.

### PR DESCRIPTION
**Pull request by cdnninja**
closes #2570.

### CODE REVIEW

**Comment by cdnninja** [comment]
@zwave-js-bot fix lint

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+				},
+				{
+					"label": "Turn on load",
+					"value": 1
+				},
+				{
+					"label": "Save load state before power failure",
+					"value": 2
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]: This comes up often enough that we should make two master templates for it. One with "previous state", one without.

IMO, this is the preferred definition (short and clear):
```suggestion
			"label": "State After Power Failure",
			"valueSize": 1,
			"minValue": 0,
			"maxValue": 2,
			"defaultValue": 2,
			"allowManualEntry": false,
			"options": [
				{
					"label": "Off",
					"value": 0
				},
				{
					"label": "On",
					"value": 1
				},
				{
					"label": "Previous State",
					"value": 2
				}
			]
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+			"options": [
+				{
+					"label": "Disable to send Basic report",
+					"value": 0
+				},
+				{
+					"label": "Enable to send Basic report",
+					"value": 1
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Send Basic Report",
			"valueSize": 1,
			"minValue": 0,
			"maxValue": 1,
			"defaultValue": 1,
			"allowManualEntry": false,
			"options": [
				{
					"label": "Disable",
					"value": 0
				},
				{
					"label": "Enable",
					"value": 1
				}
			]
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+				{
+					"label": "Disable",
+					"value": 0
+				},
+				{
+					"label": "Enable",
+					"value": 1
+				},
+				{
+					"label": "Enable to detect only after first re-power",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
					"label": "Only after first re-power",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+				},
+				{
+					"label": "Enable",
+					"value": 1
+				}
+			]
+		},
+		"10": {
+			"label": "Detect load",
+			"description": "Enable/disable to detect load",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Please give this a second pass - there are several labels and descriptions that don't align with the style guide. I've highlighted some of them.

File: packages/config/config/devices/0x0330/zv2835rac-nf.json (hunk 1 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x0330/zv2835rac-nf.json]: Once #2665 is merged, the preferred one would be
```suggestion
"$import": "../templates/master_template.json#state_after_power_failure_off_on_prev"
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+					"value": 0
+				},
+				{
+					"label": "Enable",
+					"value": 1
+				}
+			]
+		},
+		"4": {
+			"label": "Default fade time",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Default Fade Time",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+				},
+				{
+					"label": "1 to 127 minutes",
+					"value": 80
+				},
+				{
+					"label": "Factory default duration. (3 second)",
+					"value": 255
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]: I guess you're supposed to enter the values here, so this definition doesn't work. You can leave the Instantly and Factory default options. The rest with changing unit can be done as a description:
```
Values 1-127 = seconds; 128-254 = minutes (minus 127)
```
but you'll need `"allowManualEntry": true,`

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+		},
+		"5": {
+			"label": "Minimum brightness value",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Minimum Brightness Value",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 50,
+			"defaultValue": 0
+		},
+		"6": {
+			"label": "Maximum brightness value",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Maximum Brightness Value",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 100,
+			"defaultValue": 100
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json:89]: In Z-Wave the brightness (level) goes from 0 to 99, so I think this needs to be changed:
```suggestion
			"minValue": 0,
			"maxValue": 99,
			"defaultValue": 99
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+		},
+		"7": {
+			"label": "MOSFET driving type",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "MOSFET Driving Type",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+					"value": 0
+				},
+				{
+					"label": "Leading edge",
+					"value": 1
+				}
+			]
+		},
+		"8": {
+			"label": "Step of knob",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Step of Knob",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+				{
+					"label": "Disable",
+					"value": 0
+				},
+				{
+					"label": "Enable",
+					"value": 1
+				}
+			]
+		},
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]: I don't understand this param

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+		"10": {
+			"label": "Detect load",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Detect Load",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+					"value": 1
+				},
+				{
+					"label": "Only after first re-power",
+					"value": 2
+				}
+			]
+		},
+		"11": {
+			"label": "Wiring type",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Wiring Type",
```
Any explanation what these types mean?

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 2,
+			"defaultValue": 0,
+			"readOnly": true
+		},
+		"12": {
+			"label": "Load type",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Load Type",
```
Any explanation what these types mean?

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 16]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 3,
+			"defaultValue": 0,
+			"readOnly": true
+		},
+		"13": {
+			"label": "Over current protection",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Over Current Protection",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+					"value": 0
+				},
+				{
+					"label": "Enable",
+					"value": 1
+				}
+			]
+		},
+		"14": {
+			"label": "Power automatic report absolute threshold",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Automatic Power Report: Absolute Threshold",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 2,
+			"unit": "W",
+			"minValue": 0,
+			"maxValue": 100,
+			"defaultValue": 10
+		},
+		"15": {
+			"label": "Power automatic report threshold,",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Automatic Power Report: Relative Threshold",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 1,
+			"unit": "%",
+			"minValue": 0,
+			"maxValue": 100,
+			"defaultValue": 20
+		},
+		"21": {
+			"label": "Power metering time cycle",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Power Metering Time Cycle",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 4,
+			"unit": "seconds",
+			"minValue": 0,
+			"maxValue": 2678400,
+			"defaultValue": 600
+		},
+		"22": {
+			"label": "Energy metering time cycle",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Energy Metering Time Cycle",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 21]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 4,
+			"unit": "seconds",
+			"minValue": 0,
+			"maxValue": 2678400,
+			"defaultValue": 1800
+		},
+		"23": {
+			"label": "Voltage metering time cycle",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Voltage Metering Time Cycle",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 22]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 4,
+			"unit": "seconds",
+			"minValue": 5,
+			"maxValue": 2678400,
+			"defaultValue": 3600
+		},
+		"24": {
+			"label": "Current metering time cycle",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "CurrentMetering Time Cycle",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 23]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 4,
+			"unit": "seconds",
+			"minValue": 5,
+			"maxValue": 2678400,
+			"defaultValue": 3600
+		},
+		"31": {
+			"label": "Dimming curve",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Dimming Curve",
```

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 24]
```diff
@@ -0,0 +1,271 @@
...
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 0,
+			"options": [
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json:246]: This should have `allowManualEntry: false`

File: packages/config/config/devices/0x0330/zv2835rac-nf.json [hunk 25]
```diff
@@ -0,0 +1,271 @@
...
+					"value": 0
+				},
+				{
+					"label": "Logarithmic",
+					"value": 1
+				}
+			]
+		},
+		"32": {
+			"label": "Starting brightness",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2835rac-nf.json]:
```suggestion
			"label": "Starting Brightness",
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Comment by cdnninja** [comment]
@AlCalzone  is this lint error a mistake in logic on the check?

**Comment by AlCalzone** [comment]
No, its a somewhat recent change to the lint rules. `allowManualEntry` defaults to `true` for writable parameters and can now only be set to `false` or left out.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**